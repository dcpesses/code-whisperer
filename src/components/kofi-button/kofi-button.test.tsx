import {render} from '@testing-library/react';
import KofiButton from './index';

describe('KofiButton', () => {
  test('Should render as button for a link that opens in a new window', () => {
    const {container} = render(<KofiButton />);
    expect(container).toMatchSnapshot();
  });
  test('Should render as button for a link that opens in the same window', () => {
    const {container} = render(<KofiButton opensInNewWindow={false} />);
    expect(container).toMatchSnapshot();
  });
});
