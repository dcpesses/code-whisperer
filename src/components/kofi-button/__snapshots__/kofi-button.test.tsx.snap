// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`KofiButton > Should render as button for a link that opens in a new window 1`] = `
<div>
  <button
    class="kofi-button btn btn-primary btn-sm"
    type="button"
  >
    <img
      alt="Ko-fi logo"
      class="img-fluid"
      src="data:image/svg+xml,%3csvg%20width='241'%20height='194'%20viewBox='0%200%20241%20194'%20fill='none'%20xmlns='http://www.w3.org/2000/svg'%3e%3cmask%20id='mask0_1_219'%20style='mask-type:luminance'%20maskUnits='userSpaceOnUse'%20x='-1'%20y='0'%20width='242'%20height='194'%3e%3cpath%20d='M240.469%200.958984H-0.00585938V193.918H240.469V0.958984Z'%20fill='white'/%3e%3c/mask%3e%3cg%20mask='url(%23mask0_1_219)'%3e%3cpath%20d='M96.1344%20193.911C61.1312%20193.911%2032.6597%20178.256%2015.9721%20149.829C1.19788%20124.912%20-0.00585938%2097.9229%20-0.00585938%2067.7662C-0.00585938%2049.8876%205.37293%2034.3215%2015.5413%2022.7466C24.8861%2012.1157%2038.1271%205.22907%2052.8317%203.35378C70.2858%201.14271%2091.9848%200.958984%20114.545%200.958984C151.259%200.958984%20161.63%201.4088%20176.075%202.85328C195.29%204.76026%20211.458%2011.932%20222.824%2023.5955C234.368%2035.4428%20240.469%2051.2624%20240.469%2069.3627V72.9994C240.469%20103.885%20219.821%20129.733%20191.046%20136.759C188.898%20141.827%20186.237%20146.871%20183.089%20151.837L183.006%20151.964C172.869%20167.632%20149.042%20193.918%20103.401%20193.918H96.1281L96.1344%20193.911Z'%20fill='white'/%3e%3cpath%20d='M174.568%2017.9772C160.927%2016.6151%20151.38%2016.1589%20114.552%2016.1589C90.908%2016.1589%2070.9008%2016.387%2054.7644%2018.4334C33.3949%2021.164%2015.2058%2037.5285%2015.2058%2067.7674C15.2058%2098.0066%2016.796%20121.422%2029.0741%20142.107C42.9425%20165.751%2066.1302%20178.707%2096.1412%20178.707H103.414C140.242%20178.707%20160.25%20159.156%20170.253%20143.698C174.574%20136.874%20177.754%20130.058%20179.801%20123.234C205.947%20120.96%20225.27%2099.3624%20225.27%2072.9941V69.3577C225.27%2040.9432%20206.631%2021.164%20174.574%2017.9772H174.568Z'%20fill='white'/%3e%3cpath%20d='M15.1975%2067.7674C15.1975%2037.5285%2033.3866%2021.164%2054.7559%2018.4334C70.8987%2016.387%2090.906%2016.1589%20114.544%2016.1589C151.372%2016.1589%20160.919%2016.6151%20174.559%2017.9772C206.617%2021.1576%20225.255%2040.937%20225.255%2069.3577V72.9941C225.255%2099.3687%20205.932%20120.966%20179.786%20123.234C177.74%20130.058%20174.559%20136.874%20170.238%20143.698C160.235%20159.156%20140.228%20178.707%20103.4%20178.707H96.1264C66.1155%20178.707%2042.9277%20165.751%2029.0595%20142.107C16.7814%20121.422%2015.1912%2098.4563%2015.1912%2067.7674'%20fill='%23202020'/%3e%3cpath%20d='M32.2469%2067.9899C32.2469%2097.3168%2034.0654%20116.184%2043.6127%20133.689C54.5225%20153.924%2074.3018%20161.653%2096.8117%20161.653H103.857C133.411%20161.653%20147.736%20147.329%20155.693%20134.829C159.558%20128.462%20162.966%20121.417%20164.784%20112.547L166.147%20106.864H174.332C192.521%20106.864%20208.208%2092.09%20208.208%2073.2166V69.8082C208.208%2048.6669%20195.024%2037.5228%20172.058%2034.7987C159.102%2033.6646%20151.372%2033.2084%20114.538%2033.2084C89.7602%2033.2084%2072.0272%2033.4364%2058.6152%2035.4828C39.7483%2038.2134%2032.2407%2048.8951%2032.2407%2067.9899'%20fill='white'/%3e%3cpath%20d='M166.158%2083.6801C166.158%2086.4107%20168.204%2088.4572%20171.841%2088.4572C183.435%2088.4572%20189.802%2081.8619%20189.802%2070.9523C189.802%2060.0427%20183.435%2053.2195%20171.841%2053.2195C168.204%2053.2195%20166.158%2055.2657%20166.158%2057.9963V83.6866V83.6801Z'%20fill='%23202020'/%3e%3cpath%20d='M54.5321%2082.3198C54.5321%2095.732%2062.0332%20107.326%2071.5807%20116.424C77.9478%20122.562%2087.9515%20128.93%2094.7685%20133.022C96.8147%20134.157%2098.8611%20134.841%20101.136%20134.841C103.866%20134.841%20106.134%20134.157%20107.959%20133.022C114.782%20128.93%20124.779%20122.562%20130.919%20116.424C140.694%20107.332%20148.195%2095.7383%20148.195%2082.3198C148.195%2067.7673%20137.286%2054.8115%20121.599%2054.8115C112.28%2054.8115%20105.912%2059.5882%20101.136%2066.1772C96.8147%2059.582%2090.2259%2054.8115%2080.9001%2054.8115C64.9855%2054.8115%2054.5256%2067.7673%2054.5256%2082.3198'%20fill='%23FF5A16'/%3e%3c/g%3e%3c/svg%3e"
    />
    <span
      class="px-2 align-middle"
    >
      Support me on Ko-fi 
    </span>
    <i
      aria-hidden="true"
      class="bi bi-box-arrow-up-right small"
    />
    <span
      class="visually-hidden"
    >
       (link opens a new window)
    </span>
  </button>
</div>
`;

exports[`KofiButton > Should render as button for a link that opens in the same window 1`] = `
<div>
  <button
    class="kofi-button btn btn-primary btn-sm"
    type="button"
  >
    <img
      alt="Ko-fi logo"
      class="img-fluid"
      src="data:image/svg+xml,%3csvg%20width='241'%20height='194'%20viewBox='0%200%20241%20194'%20fill='none'%20xmlns='http://www.w3.org/2000/svg'%3e%3cmask%20id='mask0_1_219'%20style='mask-type:luminance'%20maskUnits='userSpaceOnUse'%20x='-1'%20y='0'%20width='242'%20height='194'%3e%3cpath%20d='M240.469%200.958984H-0.00585938V193.918H240.469V0.958984Z'%20fill='white'/%3e%3c/mask%3e%3cg%20mask='url(%23mask0_1_219)'%3e%3cpath%20d='M96.1344%20193.911C61.1312%20193.911%2032.6597%20178.256%2015.9721%20149.829C1.19788%20124.912%20-0.00585938%2097.9229%20-0.00585938%2067.7662C-0.00585938%2049.8876%205.37293%2034.3215%2015.5413%2022.7466C24.8861%2012.1157%2038.1271%205.22907%2052.8317%203.35378C70.2858%201.14271%2091.9848%200.958984%20114.545%200.958984C151.259%200.958984%20161.63%201.4088%20176.075%202.85328C195.29%204.76026%20211.458%2011.932%20222.824%2023.5955C234.368%2035.4428%20240.469%2051.2624%20240.469%2069.3627V72.9994C240.469%20103.885%20219.821%20129.733%20191.046%20136.759C188.898%20141.827%20186.237%20146.871%20183.089%20151.837L183.006%20151.964C172.869%20167.632%20149.042%20193.918%20103.401%20193.918H96.1281L96.1344%20193.911Z'%20fill='white'/%3e%3cpath%20d='M174.568%2017.9772C160.927%2016.6151%20151.38%2016.1589%20114.552%2016.1589C90.908%2016.1589%2070.9008%2016.387%2054.7644%2018.4334C33.3949%2021.164%2015.2058%2037.5285%2015.2058%2067.7674C15.2058%2098.0066%2016.796%20121.422%2029.0741%20142.107C42.9425%20165.751%2066.1302%20178.707%2096.1412%20178.707H103.414C140.242%20178.707%20160.25%20159.156%20170.253%20143.698C174.574%20136.874%20177.754%20130.058%20179.801%20123.234C205.947%20120.96%20225.27%2099.3624%20225.27%2072.9941V69.3577C225.27%2040.9432%20206.631%2021.164%20174.574%2017.9772H174.568Z'%20fill='white'/%3e%3cpath%20d='M15.1975%2067.7674C15.1975%2037.5285%2033.3866%2021.164%2054.7559%2018.4334C70.8987%2016.387%2090.906%2016.1589%20114.544%2016.1589C151.372%2016.1589%20160.919%2016.6151%20174.559%2017.9772C206.617%2021.1576%20225.255%2040.937%20225.255%2069.3577V72.9941C225.255%2099.3687%20205.932%20120.966%20179.786%20123.234C177.74%20130.058%20174.559%20136.874%20170.238%20143.698C160.235%20159.156%20140.228%20178.707%20103.4%20178.707H96.1264C66.1155%20178.707%2042.9277%20165.751%2029.0595%20142.107C16.7814%20121.422%2015.1912%2098.4563%2015.1912%2067.7674'%20fill='%23202020'/%3e%3cpath%20d='M32.2469%2067.9899C32.2469%2097.3168%2034.0654%20116.184%2043.6127%20133.689C54.5225%20153.924%2074.3018%20161.653%2096.8117%20161.653H103.857C133.411%20161.653%20147.736%20147.329%20155.693%20134.829C159.558%20128.462%20162.966%20121.417%20164.784%20112.547L166.147%20106.864H174.332C192.521%20106.864%20208.208%2092.09%20208.208%2073.2166V69.8082C208.208%2048.6669%20195.024%2037.5228%20172.058%2034.7987C159.102%2033.6646%20151.372%2033.2084%20114.538%2033.2084C89.7602%2033.2084%2072.0272%2033.4364%2058.6152%2035.4828C39.7483%2038.2134%2032.2407%2048.8951%2032.2407%2067.9899'%20fill='white'/%3e%3cpath%20d='M166.158%2083.6801C166.158%2086.4107%20168.204%2088.4572%20171.841%2088.4572C183.435%2088.4572%20189.802%2081.8619%20189.802%2070.9523C189.802%2060.0427%20183.435%2053.2195%20171.841%2053.2195C168.204%2053.2195%20166.158%2055.2657%20166.158%2057.9963V83.6866V83.6801Z'%20fill='%23202020'/%3e%3cpath%20d='M54.5321%2082.3198C54.5321%2095.732%2062.0332%20107.326%2071.5807%20116.424C77.9478%20122.562%2087.9515%20128.93%2094.7685%20133.022C96.8147%20134.157%2098.8611%20134.841%20101.136%20134.841C103.866%20134.841%20106.134%20134.157%20107.959%20133.022C114.782%20128.93%20124.779%20122.562%20130.919%20116.424C140.694%20107.332%20148.195%2095.7383%20148.195%2082.3198C148.195%2067.7673%20137.286%2054.8115%20121.599%2054.8115C112.28%2054.8115%20105.912%2059.5882%20101.136%2066.1772C96.8147%2059.582%2090.2259%2054.8115%2080.9001%2054.8115C64.9855%2054.8115%2054.5256%2067.7673%2054.5256%2082.3198'%20fill='%23FF5A16'/%3e%3c/g%3e%3c/svg%3e"
    />
    <span
      class="px-2 align-middle"
    >
      Support me on Ko-fi 
    </span>
  </button>
</div>
`;
